lib/plugins/filter/after_render/injector.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const { Cache } = require('hexo-util');
+const cache = new Cache();
+
+function injectFilter(data, locals = { page: {} }) {
+  const { injector: _Injector } = this.extend;
+
+  const current = () => {
+    const { page } = locals;
+
+    if (page.__index) return 'home';
+    if (page.__post) return 'post';
+    if (page.__page) return 'page';
+    if (page.archive) return 'archive';
+    if (page.category) return 'category';
+    if (page.tag) return 'tag';
+    return 'default';
+  };
+
+  function injector(data, pattern, flag, isBegin = true) {
+    if (data.includes(`hexo injector ${flag}`)) return data;
+
+    const arr = cache.apply(
+      `${flag}-${current()}-arr`,
+      _Injector.get(flag).filter(i => i.to === current()).map(i => i.value)
```

Comment:
The size of `arr` is not large, the use of cache may not have much improvement.

Reply:
After enabling cache, benchmark (performed by travis ci) shows the time drops from `16.3s` to `15.8s` on average.

It is not about the size of array, it is about how many times the `injector()` will be executed.